test/e2e/alertmanager_test.go

```diff
@@ -135,6 +135,130 @@ enableUserWorkload: true`,
 	}
 }
 
+func TestAlertmanagerPolicyHeaders(t *testing.T) {
```

Comment:
(suggestion) I would group the checks into one test with sub-tests (one per component). IMHO it will be easier to maintain and share code.

Reply:
Updated, pleas tell me if now looks better